File: array/1608_special_array_with_x_elements_greater_than_or_equal_x/1608_special_array_with_x_elements_greater_than_or_equal_x.py
from typing import List

class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        left, right = 0, len(nums)
        while left < right:
            mid = left + (right - left) // 2
            if mid < nums[mid]:
                left = mid + 1
            else:
                right = mid
        return -1 if left < len(nums) and left == nums[left] else left


nums = [3,5]
res = Solution().specialArray(nums)
print(res)

File: array/1608_special_array_with_x_elements_greater_than_or_equal_x/1608_special_array_with_x_elements_greater_than_or_equal_x.go
package main

import "fmt"

func specialArray(nums []int) int {

	counting := make([]int, 1001)
	for i := range nums {
		counting[nums[i]]++
	}

	for i := len(counting) - 2; i > -1; i-- {
		counting[i] += counting[i+1]
	}

	for i := 0; i <= len(nums); i++ {
		if counting[i] == i {
			return i
		}
	}

	return -1
}

func main() {
	nums := []int{3, 5}
	res := specialArray(nums)
	fmt.Println(res)
}
